spark/src/main/scala/org/apache/spark/sql/delta/actions/TableFeatureSupport.scala

```diff
@@ -258,24 +270,32 @@ trait TableFeatureSupport { this: Protocol =>
   def canDowngradeTo(to: Protocol, droppedFeatureName: String): Boolean = {
     if (!supportsWriterFeatures) return false
 
-    // When `to` protocol does not have any features version downgrades are possible. However,
+    val featureNames = readerAndWriterFeatureNames - droppedFeatureName
+    val minRequiredVersions = TableFeatureProtocolUtils.minimumRequiredVersions(
+      featureNames.flatMap(TableFeature.featureNameToFeature).toSeq)
+    val toSupportsRequiredMinVersions =
+      (to.minReaderVersion, to.minWriterVersion) == minRequiredVersions
+
+    // When `to` protocol does not have any features, versions might get downgraded. However,
     // the current protocol needs to contain one non-legacy feature. We also allow downgrade when
     // there are only legacy features. This is to accommodate the case when the user attempts to
     // remove a legacy feature in a table that only contains legacy features.
```

Comment:
This comment is complicated. It distinguishes between only-legacy-features and at-least-one-non-legacy-feature but the calculation is really simple: the table after downgrade needs to have the right set of features.
```suggestion
    // If the target protocol doesn't have any explicit table features and only uses
    // legacy protocol versions, then we can downgrade if the downgraded protocol
    // has the implicitly supported features that we need, i.e., the current protocol's
    // features minus the dropped feature.
```